examples/tts/magpietts.py

```diff
@@ -30,12 +30,21 @@
 @hydra_runner(config_path="conf/magpietts", config_name="magpietts_en")
 def main(cfg):
     logging.info('\nConfig Params:\n%s', OmegaConf.to_yaml(cfg, resolve=True))
-    if not cfg.model.get('use_lthose', False):
-        import torch.multiprocessing as mp
 
-        mp.set_start_method("spawn", force=True)
+    # forcing "spawn" method for multiprocessing over "fork" when choosing multiple
+    # worker processes for dataloaders. By default, multiprocessing uses "fork" to create
+    # worker processes, which inherit the memory state of the main process, including its
+    # already initialized CUDA state. When the worker processes trieds to use
+    # CUDA, it runs into conflicts with the inherited, now potentially invalid,
+    # CUDA context, resuling in the CUDA initialization error. When
+    # num_workers=0, all dataloading happens in the main process, so there is no
+    # process forking and no CUDA context conflict. When num_workers>0, the standard way
+    # to fix this is to use "spawn" to create a completely new and clean python process for
+    # each worker, avoding the problematic CUDA state inheritance.
+    mp.set_start_method("spawn", force=True)
 
     trainer = pl.Trainer(**cfg.trainer)
+    trainer.callbacks.append(pl.callbacks.LearningRateMonitor(logging_interval='step', log_weight_decay=True))
```

Comment:
What does this do?

Reply:
this is used to log LR. add this built-in callback to trigger automated LR logging. So we don't need to manually fetch and log,
```python
lr = self.optimizers().param_groups[0]['lr']  # Manually fetch LR
self.log('learning_rate', lr, on_step=True, on_epoch=True, prog_bar=True, logger=True)
```